Rules/UseUsingScopeModifierInNewRunspaces.cs

```diff
@@ -0,0 +1,375 @@
+        /// <param name="fileName">Name of file containing the ast</param>
+        /// <returns>An enumerable containing diagnostic records</returns>
+        private IEnumerable<DiagnosticRecord> AnalyzeScriptBlockAst(ScriptBlockAst scriptBlockAst, string fileName)
+        {
+            var astsToProcess = new List<Ast>();
+            var nonAssignedNonUsingVars = new List<VariableExpressionAst>();
+
+            astsToProcess.AddRange(
+                FindAllInlineScriptAsts(scriptBlockAst));
+
+            astsToProcess.AddRange(
+                FindAllStartJobAsts(scriptBlockAst));
+
+            astsToProcess.AddRange(
+                FindAllForeachParallelAsts(scriptBlockAst));
+
+            //   If -ComputerName or -Session is not specified, you cannot use a variable with a using: scope modifier. 
+            //   Also tricky; one can open a persistent session, and use subsequent `invoke-command` calls to that, where variables
+            //   assigned in previous calls are still valid:
+            //   https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_remote_variables?view=powershell-7#using-remote-variables
+            //   Because -ComputerName and -Session parameters are in different parameter sets, we can treat them separately
+            astsToProcess.AddRange(
+                FindAllInvokeCommandComputerAsts(scriptBlockAst));
+
+            // process Invoke-Command -Session separately
+            nonAssignedNonUsingVars.AddRange(
+                ProcessInvokeCommandSessionAsts(scriptBlockAst));
+
+
+            foreach (var ast in astsToProcess)
+            {
+                if (ast == null)
+                    continue;
+
+                var varsInAssignments = FindVarsInAssignmentAsts(ast);
+                
+                nonAssignedNonUsingVars.AddRange(
+                    FindNonAssignedNonUsingVarAsts(ast, varsInAssignments));
+            }
+            
+            foreach (var variableExpression in nonAssignedNonUsingVars)
+            {
+                if (variableExpression == null)
+                    continue;
+
+                yield return new DiagnosticRecord(
+                    message: string.Format(CultureInfo.CurrentCulture,
+                        Strings.UseUsingScopeModifierInNewRunspacesError, variableExpression.ToString()),
+                    extent: variableExpression.Extent,
+                    ruleName: GetName(),
+                    severity: DiagnosticSeverity.Warning,
+                    scriptPath: fileName,
+                    ruleId: variableExpression.ToString(),
+                    suggestedCorrections: GetSuggestedCorrections(ast: variableExpression));
+            }
+        }
+
+        private static IEnumerable<CorrectionExtent> GetSuggestedCorrections(Ast ast)
+        {
+            var varWithUsing = "$using:" + ast.Extent.Text.TrimStart('$');

```

Comment:
Rather than use `ast.Extent.Text` you should use `ast.VariablePath.UserPath`:

```suggestion
        private static IEnumerable<CorrectionExtent> GetSuggestedCorrections(VariableExpressionAst ast)
        {
            var varWithUsing = $"$using:{ast.VariablePath.UserPath}";

```